test-data/unit/check-based-default-return.test

```diff
@@ -0,0 +1,37 @@
+-- Type checker test cases for default return type.
+
+[case testDefault]
+# based
+def f(): ...
+reveal_type(f)  # N: Revealed type is "def ()"
+
+def g(i: int): ...
+reveal_type(g)  # N: Revealed type is "def (i: builtins.int)"
+
+[case testUntypedDefs]
+# flags: --allow-untyped-defs --allow-any-expr
+
+def f(): ...
+reveal_type(f)  # N: Revealed type is "def () -> Any"
+
+def g(i: int): ...
+reveal_type(g)  # N: Revealed type is "def (i: builtins.int)"
+
+[case testIncompleteDefs]
+# flags: --allow-incomplete-defs --allow-untyped-defs --allow-any-expr
+
+def f(i): ...
+reveal_type(f)  # N: Revealed type is "def (i: Any) -> Any"
+
+def g(i: int, j): ...
+reveal_type(g)  # N: Revealed type is "def (i: builtins.int, j: Any)"
+
+[case testNoDefaultReturn]
+# flags: --no-default-return  --allow-any-expr
+
+def f(i): ...  # E: Function is missing a type annotation  [no-untyped-def]
+reveal_type(g)  # N: Revealed type is "def (i: builtins.int, j: Any) -> Any"
+
+def g(i: int, j): ...  # E: Function is missing a return type annotation  [no-untyped-def] \
+    # E: Function is missing a type annotation for one or more arguments  [no-untyped-def]
+reveal_type(g)  # N: Revealed type is "def (i: builtins.int, j: Any) -> Any"
```

Comment:
maybe add tests for other types of functions like lambdas and methods?

Reply:
and generators too